src/app/plugins/addroles.plugin.ts

```diff
@@ -33,22 +42,8 @@ export class AddRolesPlugin extends Plugin {
         await member.roles.add(role);
         roles_added.push(role.name);
 
-        if (role.name.toLowerCase() === 'alumni') {
-          const boomerEmoji = this.container.guildService.get().emojis.cache
-          .filter(n => n.name.toLowerCase() === 'okboomer').first();
-
-          if (boomerEmoji) {
-            message.react(boomerEmoji);
-          }
-        }
-        else if (role.name.toLowerCase() === 'gradstudent') {
-          const knightEmoji = this.container.guildService.get().emojis.cache
-            .filter(n => n.name.toLowerCase() === 'knight').first();
-  
-          if (knightEmoji) {
-            message.react(knightEmoji);
-          }
-        }
+        if (role.name.toLowerCase() === 'alumni') this.react('okboomer', message);
+        else if (role.name.toLowerCase() === 'gradstudent') this.react('knight', message);
```

Comment:
handle the `if` logic in the `react` func instead of out here

Reply:
Cache the emojis as a property as like a record so you have
```
private emojis: Record<string, IRoleEmoji> = {
    alumni: { emojiName: 'knight', emoji: null },
    gradstudent: { emojiName: 'okboomer', emoji: null },
};
```
and assign them in the constructor that way we dont have to find it, and make sure it exists each time

then you can do something like
```
await Promise.all(
  Object.keys(this.emojis).map((key) => {
   if (role.toLowerCase() !== key) {
      return;
   }

   return message.react(this.emojis[key].emojiName);
  })
);
```
That way its easily scalable

Sorry if that not correct

Reply:
also make sure you are awaiting promises

Reply:
for that example `IRoleEmoji` would be a simple interface with a string for emojiName and ReactionEmoji, which is the actual emoji

Reply:
This tbh